benchmarks/dynamo/common.py

```diff
@@ -786,6 +788,22 @@ def inner(*args, **kwargs):
 
     return inner
 
+@contextmanager
+def maybe_init_distributed(should_init_distributed, rank=0, world_size=1):
+    # To avoid multiple inheritance from _dynamo.test_case.TestCase and MultiProcessTestCase,
+    # Just manually implement the most important part of the dynamo behavior to reset/clear.
+    try:
+        if should_init_distributed:
+            torch.cuda.set_device(rank)
+            os.environ['MASTER_ADDR'] = 'localhost'
+            os.environ['MASTER_PORT'] = '6789'
```

Comment:
Will this break if 2 instances of this benchmark are running on the same machine at once?

Reply:
yes. :(

is there a good pattern i can copy for getting a free port?